tests/integration/relationships/belongs-to-test.js

```diff
@@ -587,6 +587,524 @@ test('The store can load a polymorphic belongsTo association', function(assert)
+  env.registry.register('model:vetenarian', Vetenarian);
+  env.registry.register('model:pet', Pet);
+
+  env.registry.register(
+    'adapter:vetenarian',
+    DS.JSONAPIAdapter.extend({
+      findRecord(store, type, id) {
+        return resolve({
+          data: {
+            type: 'person',
+            id,
+            relationships: {
+              pets: {
+                data: [
+                  {
+                    type: 'pet',
+                    id: '2',
+                  },
+                ],
+              },
+            },
+          },
+        });
+      },
+    })
+  );
+
+  env.registry.register(
+    'adapter:pet',
+    DS.JSONAPIAdapter.extend({
+      findRecord(store, type, id) {
+        return resolve({
+          data: {
+            type: 'pet',
+            id,
+            relationships: {
+              human: {
+                link: {
+                  type: 'vetenarian',
+                  id: '1',
+                },
+              },
+            },
+          },
+        });
+      },
+    })
+  );
+  let pet = await store.findRecord('pet', 2);
+  let person = await pet.get('human');
+  assert.ok(person);
+  assert.equal(person.id, 1);
+});
+
+test('The store can load an 1-many async belongsTo association with no inverses via json-api', async function(assert) {
+  class Person extends Model {
+    @belongsTo('pet', { inverse: null, async: true })
+    bestDog;
+    @attr
+    name;
+  }
+
+  class Pet extends Model {
+    @hasMany('person', { inverse: null, async: true })
+    humans;
+    @attr
+    name;
+  }
+
+  env.registry.register('model:person', Person);
+  env.registry.register('model:pet', Pet);
+
+  env.registry.register(
+    'adapter:person',
+    DS.JSONAPIAdapter.extend({
+      findRecord(store, type, id) {
+        return resolve({
+          data: {
+            type: 'person',
+            id,
+            relationships: {
+              'best-dog': {
+                data: {
+                  type: 'pet',
+                  id: '2',
+                },
+              },
+            },
+          },
+        });
+      },
+    })
+  );
+
+  env.registry.register(
+    'adapter:pet',
+    DS.JSONAPIAdapter.extend({
+      findRecord(store, type, id) {
+        return resolve({
+          data: {
+            type: 'pet',
+            id,
+          },
+        });
+      },
+    })
+  );
+  let person = await store.findRecord('person', 1);
+  let pet = await person.get('bestDog');
+  assert.ok(pet);
+  assert.equal(pet.id, 2);
+});
+
+test('The store can load a 1-many polymorphic sync belongsTo association via json-api', async function(assert) {
+  env.registry.register(
+    'adapter:comment',
+    DS.JSONAPIAdapter.extend({
+      findRecord(store, type, id) {
+        return resolve({
+          data: {
+            type: 'comment',
+            id: '2',
+            relationships: {
+              message: {
+                data: {
+                  type: 'post',
```

Comment:
@runspired @BryanCrotaz I don't know if those tests are still somewhat valid, but here, I think there is a mistake (I've not read the whole file), the type seems completely different from the relationship name. Or maybe it's something intented, and defined as this in the model